sky/backends/cloud_vm_ray_backend.py

```diff
@@ -591,25 +611,36 @@ def add_epilogue(self) -> None:
 
         self._code += [
             textwrap.dedent(f"""\
-            returncodes = ray.get(futures)
+            returncodes = wait_or_fail(futures)
+            # Remove the placement group after all tasks are done, so that the
+            # next job can be scheduled on the released resources immediately.
+            ray_util.remove_placement_group(pg)
```

Comment:
maybe: let wait_or_fail() do this line?

Reply:
Moved it to `get_or_fail` function. Thanks!